test/unit/command_test.rb

```diff
@@ -66,5 +66,54 @@ class CommandTest < ActiveSupport::TestCase
       end
       assert_equal '/etc/passwd: Permission denied', error.message
     end
+
+    test 'sets code and message correctly on success' do
+      command = Command.new('true', chdir: '.')
+      assert_nil command.code
+      command.run
+      refute_predicate command, :running?
+      assert_predicate command.code, :zero?
+      assert_equal 'terminated successfully', command.termination_status
+    end
+
+    test 'sets code and message correctly on error' do
+      command = Command.new('false', chdir: '.')
+      assert_nil command.code
+      command.run
+      refute_predicate command, :running?
+      assert_predicate command.code, :nonzero?
+      assert_equal 'terminated with exit status 1', command.termination_status
+    end
+
+    test 'handles externally signalled commands correctly' do
+      command = Command.new('sleep 10', chdir: '.')
+      t = Thread.new do
+        signalled = false
+        20.times do
+          if command.running?
+            Process.kill('KILL', command.pid)
+            signalled = true
+            break
+          end
+          sleep 0.1
+        end
+        signalled
+      end
+      command.run
+      assert t.join, "subprocess wasn't signalled"
+      assert_predicate command, :signaled?
+      refute_predicate command, :running?
+      assert_nil command.code
+      assert_equal 'terminated with KILL signal', command.termination_status
+    end
+
+    test 'reports timedout command correctly' do
+      command = Command.new('sleep 10', chdir: '.', default_timeout: 0.5)
+      assert_raises(Command::TimedOut) { command.run }
+      assert_predicate command, :signaled?
+      refute_predicate command, :running?
+      assert_nil command.code
+      assert_equal 'terminated with INT signal', command.termination_status
```

Comment:
Is this the message that eventually gets sent to the user?

I kind of like keeping track whether *we* sent the signal to the process, because that allows us to say that *we* killed the process (due to a timeout). Otherwise I am afraid this will raise questions from people.

Reply:
> Is this the message that eventually gets sent to the user?

It is, but it's not the only output - in the case of a timeout Shipit already prints "No output received in the last _n_ seconds", though I guess that too is ambiguous. Could append that with ", timing out command" or something.

Reply:
For comparison, this is how Shopify Build does this:

- Timeout implementation: https://github.com/Shopify/shopify-build/blob/master/lib/shopify_build/command.rb#L202-L211
- Message: https://github.com/Shopify/shopify-build/blob/master/lib/shopify_build/command.rb#L170-L184

Reply:
The timeout model here is a bit different - it's not based on process execution time, but instead time without any output - so it's not directly applicable. However I think I can improve the messaging regardless 👍